File: my-submissions/m2149.py
# https://leetcode.com/problems/rearrange-array-elements-by-sign/description/


class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        output = [0] * len(nums)
        outputIndex = 0

        pos, neg = 0, 0
        while outputIndex < len(nums) :
            while nums[pos] < 0 :
                pos += 1
            while nums[neg] > 0 :
                neg += 1
            
            output[outputIndex] = nums[pos]
            output[outputIndex + 1] = nums[neg]
            outputIndex += 2
            pos += 1
            neg += 1
        return output
            

File: my-submissions/m2149.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* rearrangeArray(int* nums, int numsSize, int* returnSize) {
    int* output = (int*) malloc(sizeof(int) * numsSize);
    int outputCurrent = 0;
    *returnSize = numsSize;

    int neg = 0;
    int pos = 0;
    while (outputCurrent < numsSize) {
        while (nums[neg] > 0) {
            neg++;
        }
        while (nums[pos] < 0) {
            pos++;
        }

        output[outputCurrent] = nums[pos];
        output[outputCurrent + 1] = nums[neg];
        outputCurrent += 2;
        neg++;
        pos++;
    }

    return output;
}